src/main/java/io/appium/java_client/AppiumDriver.java

```diff
@@ -88,46 +88,94 @@ public AppiumDriver(ClientConfig clientConfig, Capabilities capabilities) {
         this(new AppiumCommandExecutor(MobileCommand.commandRepository, clientConfig), capabilities);
     }
 
+    public AppiumDriver(ClientConfig clientConfig, Capabilities capabilities, AppiumClientConfig appiumClientConfig) {
```

Comment:
There are too many permutations. Lets only have one constructor with all possible arguments and add other if this is really needed/requested